In [1]:
for i in range(5):
    print(i)

0
1
2
3
4


In [2]:
for i in range(1,5,2):
    print(i)

1
3


In [26]:
student = {'name':'hung',
           'degree': 'master'}
for key, value in student.items():
    print(f"Day la {key}:{value}")

Day la name:hung
Day la degree:master


In [28]:
i = 0
while i<10:
    print(i)
    i += 1

0
1
2
3
4
5
6
7
8
9


In [2]:
i = 0
while True:
    print = i
    i += 1
    if(i>10):
        break

W4A1
[SumOfNumBers]
Viết chương trình Python sử dụng một trong 2 loại vòng lặp for hoặc while để
tính tống các số nguyên từ 1 đến n. (n <= 1000)

In [3]:
n = int(input())
tong = 0
if n<=1000:
    for i in range(1, n+1):
        tong += i
    print(tong)
else:
    print("n phai nho hon 1000")

m = n
tong = 0
while 0<=m<=1000:
    tong += m
    m -= 1
print(tong)


15
15


W4A2
[PrimeNumber]
VIết chương trình nhập vào một số tới khi được một số nguyên dương thì thôi.
Kiểm tra số này có phải là số nguyên tố hay không?

In [7]:
n = int(input())
if n<=1:
    print(f"{n} khong phai la so nguyen to")
else:
    huyen = 0
    for i in range(1,n+1):
        if n%i == 0:
            huyen += 1
    if huyen == 2:
        print(f"{n} la so nguyen to")
    else:       
        print(f"{n} khong phai la so nguyen to")

8 khong phai la so nguyen to


In [16]:
import math
n = int(input())
if n<=1:
    print(f"{n} khong phai la so nguyen to")
else:
    for i in range(2,math.isqrt(n) + 1):
        if n%i == 0:
            print(f"{n} la hop so")
            break
    else:
        print(f"{n} la so nguyen to")

234 la hop so


W4A3
[Factorial]
Viết chương trình tính giai thừa của một số nguyên n cho trước (n!=1×2×…×n)
với 0 < n < 100.

In [20]:
n = int(input())
giai_thua = 1

if 0 < n < 100:
    i = 1
    while i <= n:
        giai_thua *= i
        i += 1
    print(f"Giai thua cua {n} la {giai_thua}")
else:
    print("n khong hop le")

Giai thua cua 3 la 6


W4A4
.[CountingDigits]
Viết chương trình nhập vào một số nguyên n và in ra màn hình số chữ số (trừ dấu)
của số đó. Không dùng kiểu dữ liệu string.

In [4]:
n = abs(int(input()))
count = 0
if n == 0:
    count = 1
else:
    while n > 0:
        n //= 10
        count += 1
print(count)

14


W4A5 [MeaningOfLife]
Khi Douglas Adam viết cuốn sách, The Hitchhiker’s Guide to the Galaxy, ông ta đã
thêm vào đó một vài lời nói đùa, và theo năm tháng nó đã trở nên nổi tiếng hơn cả
bản thân cuốn tiểu thuyết: “Câu trả lời cho câu hỏi lớn nhất về sự sống, về vũ trụ và
tất cả mọi thứ là 42.” Một vài người cuồng tín đã tốn hàng năm trời, với mọi cố
gắng nỗ lực để gán ghép ý nghĩa, biểu trưng cho con số 42 và các “sự cố” liên quan
đến nó.
Với mong muốn hái ra tiền từ sư “ám ảnh” điên rồ này, một cuốn sách mới vừa
được xuất bản, 42: Douglas Adams’ Amazingly Accurate Answer to Life, the
Universe and Everything. Cuốn sách được phát hành trùng với kỉ niệm 10 năm ngày
mất của Adams.
“Câu trả lời rất đơn giản”, như Adams đã nói, “Chỉ là một trò đùa. Tôi cần một con
số, một con số be bé, một con số tầm thường, và tôi đã chọn nó. Hai chữ số, bội của
13, một con số hoàn toàn vô nghĩa đối với mấy tay thầy chùa. Tôi ngồi trên bàn làm
việc, ngắm khu vườn và nghĩ rằng số 42 sẽ được việc lắm đây. Tôi gõ nó ra. Và
hết.”
Xuyên suốt lịch sử, các con số được gán cho những ý nghĩa đặc biệt nào đó. Plato
gọi việc nghiên cứu các biểu tượng số học” là level cao nhất của tri thức. Trong khi
Pythagoras tin rằng các con số đều có linh hồn cũng như sức mạnh kì diệu.
Trong khi đó, hàng triệu người hâm hộ Hitchhiker cho đến ngày nay vẫn kiên trì
giải mã những thứ họ cho rằng đó là những động cơ bí mật của Adams.
Hãy viết một chương trình nhận vào một số nguyên dương n và một dãy gồm n số
nguyên. Nếu n số nguyên đó có số 42 thì in ra màn hình dòng chữ "I′ve found the
meaning of life!", ngược lại in ra dòng chữ "It′s a joke!".

In [7]:
n = int(input())
found = False

numbers = list(map(int, input().split()))

for x in numbers:
    if x == 42:
        found = True
        break

if found:
    print("I've found the meaning of life!")
else:
    print("It's a joke!")


I've found the meaning of life!


In [8]:
numbers = list(map(int, input().split()))

if 42 in numbers:
    print("I've found the meaning of life!")
else:
    print("It's a joke!")


I've found the meaning of life!


W4A6
"[SumPrimesInRange]
Nhập hai số nguyên dương a, b (a ≤ b). Hãy tính tổng các số nguyên tố trong đoạn
[a, b]. Mỗi dòng là một bộ test (a b)."

In [2]:
import math

def is_prime(n):
    if n < 2:
        return False
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True

try:
    line = input().split()
    if len(line) == 2:
        a = int(line[0])
        b = int(line[1])
        sum = 0
    for i in range(a, b+1):
        if is_prime(i):
            sum += i
    print(sum)
except EOFError:
    pass

220


W4A7
"[LargestPrimeFactor]
Nhập vào số nguyên dương n (n ≥ 2). Hãy in ra ước số nguyên tố lớn nhất của n."

In [13]:
import math

n = int(input())
max_prime = -1

while n % 2 == 0:
    max_prime = 2
    n = n // 2

i = 3
while i * i <= n:
    while n % i == 0:
        max_prime = i
        n = n // i
    i += 2

if n > 1:
    max_prime = n

print(max_prime)


43


W4A8
ReverseAddPalindrome]
Nhập số nguyên dương n. Lặp lại phép toán n = n + reverse(n) cho đến khi n là số
đối xứng (palindrome). In ra số bước và giá trị palindrome thu được.

In [ ]:
n_str = input()
n = int(n_str)
steps = 0

while str(n) != str(n)[::-1]:
    reversed_n = int(str(n)[::-1])
    n = n + reversed_n
    steps += 1

print(steps, n)

0 5665


W4A9
[PerfectSquaresDistinctDigits]
Nhập vào số nguyên dương n. In ra tất cả các số chính phương ≤ n có các chữ số
đôi một khác nhau (in theo thứ tự tăng dần, cùng hàng, cách nhau bởi một dấu
cách). Nếu không có, in ra (no number).

In [30]:
n_str = input()
if n_str:
    n = int(n_str)
    results = []
    i = 0
    
    # Tìm các số chính phương <= n
    while True:
        square = i * i
        if square > n:
            break
        
        # Kiểm tra các chữ số đôi một khác nhau
        s = str(square)
        is_distinct = True
        # Duyệt qua từng chữ số để kiểm tra trùng lặp
        for char in s:
            if s.count(char) > 1:
                is_distinct = False
                break
        
        if is_distinct:
            results.append(str(square))
        i += 1

    # In kết quả theo định dạng yêu cầu
    if len(results) == 0:
        print("(no number)")
    else:
        print(" ".join(results))



0 1 4 9 16 25 36 49 64 81 169 196 256 289 324 361 529


W4A10
"[LongestCollatzUpToN]
Nhập vào số nguyên dương n. Tìm số x trong [1, n] có độ dài dãy Collatz lớn nhất
và in ra x cùng độ dài dãy tương ứng (x L). Nếu có nhiều số cùng độ dài, chọn số
nhỏ nhất."

In [16]:
n = int(input())

max_length = 0
best_x = 0

for x in range(1, n + 1):
    current_n = x
    length = 1
    while current_n != 1:
        if current_n % 2 == 0:
            current_n = current_n // 2
        else:
            current_n = 3 * current_n + 1
        length += 1
    if length > max_length:
        max_length = length
        best_x = x

if n > 0:
    print(f"{best_x} {max_length}")

27 112


W4A11
[EvenDivisors]
Viết chương trình đếm số lượng ước số chẵn của số nguyên dương n với n < 10^6.

In [29]:
import math

n = int(input())

count = 0

for i in range(1, int(math.sqrt(n)) + 1):
    if n % i == 0:
        if i % 2 == 0:
            count += 1

        j = n // i
        if j != i:
            if j % 2 == 0:
                count += 1

print(count)



8


W4A12
[Interest]
Một người có tài khoản tiết kiệm ở ngân hàng và gửi vào X đồng với lãi suất là
0.7% mỗi tháng. Viết chương trình tính số tiền sau N tháng người ấy rút được (cả
gốc và lãi, bỏ qua phần lẻ thập phân).

In [ ]:
data = input().split()
x = float(data[0])
n = int(data[1])

tong_tien = x

for i in range(n):
    tong_tien = tong_tien * 1.007
    
print(int(tong_tien))


W4A13
[ClosePairNumber]
Hai số nguyên dương tạo thành một cặp số thân thiết khi chúng tuân theo quy luật:
Số này bằng tổng tất cả các ước của số kia (trừ chính nó) và ngược lại.
Ví dụ 220 và 284 là một cặp số thân thiết
220 ngoài chính nó ra, nó còn có 11 ước số khác là: 1, 2, 4, 5, 10, 11, 20, 44, 55,
110. Tổng của 11 ước này cho chúng ta 284
284 ngoài bản thân nó, còn có 5 ước số khác là: 1, 2, 4, 71, 142. Tổng của chúng
vừa đúng bằng 220
Nhập hai số từ bàn phím và kiểm tra xem chúng có phải là cặp số thân thiết không.
Nếu có, in ra màn hình "true", ngược lại, in ra "false"

In [19]:
m, n = map(int, input().split())
tong_uoc_m = 0
for i in range(1, m):
    if m % i == 0:
        tong_uoc_m += i

tong_uoc_n = 0
for i in range(1, n):
    if n % i == 0:
        tong_uoc_n += i

if tong_uoc_m == n and tong_uoc_n == m:
    print("true")
else:
    print("false")


false


W4A14
[GCD]
Cho hai số m và n là hai số nguyên dương nhập từ bàn phím. Viết chương trình
tìm ước chung lớn nhất của hai số m và n.

In [23]:
m, n = map(int, input().split())

a = abs(m)
b = abs(n)

if a == 0 and b == 0:
    print(0)
else:
    while b != 0:
        a, b = b, a % b
    
    print(a)


30


W4A15
[ChickenAndDog]
"Vừa gà vừa chó,
Bó lại cho tròn,
Ba mươi sáu con,
Một trăm chân chẵn"
Từ bài toán dân gian trên, mở rộng thành chương trình nhận đầu vào là tổng số
con và tổng số chân của gà và chó. Nếu tìm được số phù hợp, in ra số lượng gà và
số lượng chó. Ngược lại, in ra "invalid".

In [ ]:
con, chan = map(int, input().split())
found = False
for i in range(con + 1):
    for j in range(chan + 1):
        if (2 * i + 4 * j) == chan:
            print(i, j)
            found = True
            break
    if found:
        break
if not found:
    print("invalid")

1 16


tìm số chính phương

In [5]:
n = int(input())
t = 1
while (t*t <= n):
    x = t*t
    t += 1
    lst = [0,0,0,0,0,0,0,0,0,0]
    flag = True
    y = x
    while (y > 0):
        đv = y % 10
        y //= 10
        if lst[đv] > 1:
            flag = False
            break
    if (flag == True):
        print(x, end = " ")

1 4 9 16 25 36 49 64 

In [14]:
n = int(input())
cnt = 0
for i in range(2, n+1, 2):
    if (n%i == 0):
        print(i, end=" ")
        cnt += 1
print()
print(cnt)

2 4 8 16 82 164 328 656 
8


ước số nguyên tố lớn nhất

In [30]:
n = int(input())
for x in range(n, 1, -1):
    flag = False
    if n % x == 0:
        for i in range(2, int(x**0.5) + 1):
            if x % i == 0:
                break
        else:
            print(x)
            flag = True
    if flag == True:
        break

607


w4a16
In các số chẵn chia hết cho 3 bé hơn 100


In [1]:
for i in range(100):
    if i%2==0 and i%3==0:
        print(i, end=" ")

0 6 12 18 24 30 36 42 48 54 60 66 72 78 84 90 96 

w4a17
 Nhập vào số nguyên dương a, in ra bảng cửu chương của a 


In [3]:
a = int(input())

print(f"--- Bảng cửu chương {a} ---")

for i in range(1, 11):
    print(f"{a} x {i} = {a * i}")

--- Bảng cửu chương 1 ---
1 x 1 = 1
1 x 2 = 2
1 x 3 = 3
1 x 4 = 4
1 x 5 = 5
1 x 6 = 6
1 x 7 = 7
1 x 8 = 8
1 x 9 = 9
1 x 10 = 10


w4a18
Nhập vào số nguyên dương a và b, in toàn bộ ước chung của a và b 

In [4]:
a = int(input())
b = int(input())

n = a if a < b else b

for i in range(1, n + 1):
    if a % i == 0 and b % i == 0:
        print(i)

1
2
4


w4a19
[EvenNumBers]
Viết chương trình Python sử dụng một trong 2 loại vòng lặp for hoặc while để in ra các số
dương chẵn nhỏ hơn hoặc bằng n (n <= 1000) 


In [6]:

for i in range(0, 1001, 2):
    print(i)

0
2
4
6
8
10
12
14
16
18
20
22
24
26
28
30
32
34
36
38
40
42
44
46
48
50
52
54
56
58
60
62
64
66
68
70
72
74
76
78
80
82
84
86
88
90
92
94
96
98
100
102
104
106
108
110
112
114
116
118
120
122
124
126
128
130
132
134
136
138
140
142
144
146
148
150
152
154
156
158
160
162
164
166
168
170
172
174
176
178
180
182
184
186
188
190
192
194
196
198
200
202
204
206
208
210
212
214
216
218
220
222
224
226
228
230
232
234
236
238
240
242
244
246
248
250
252
254
256
258
260
262
264
266
268
270
272
274
276
278
280
282
284
286
288
290
292
294
296
298
300
302
304
306
308
310
312
314
316
318
320
322
324
326
328
330
332
334
336
338
340
342
344
346
348
350
352
354
356
358
360
362
364
366
368
370
372
374
376
378
380
382
384
386
388
390
392
394
396
398
400
402
404
406
408
410
412
414
416
418
420
422
424
426
428
430
432
434
436
438
440
442
444
446
448
450
452
454
456
458
460
462
464
466
468
470
472
474
476
478
480
482
484
486
488
490
492
494
496
498
500
502
504
506
508
510
512
514
516
518
520
522
524
526

W4A20 Nhập vào một số nguyên dương n, kiểm tra xem n có phải là số dạng 2^k hay không

In [10]:
n = int(input())

if n > 0 and (n & (n - 1)) == 0:
    print(f"{n} là số dạng 2^k")
else:
    print(f"{n} không phải là số dạng 2^k")

8 là số dạng 2^k


W4A21 Nhập vào số nguyên dương n, tính tổng các chữ số của n

In [ ]:
n = input()
s = 0
for i in n:
    s += int(i)
print(s)

W4A22 Nhập vào số nguyên dương n, đếm xem n có bao nhiêu chữ số chẵn, bao nhiêu chữ số lẻ

In [ ]:
n = input()
chan = 0
le = 0

for i in n:
    if int(i) % 2 == 0:
        chan += 1
    else:
        le += 1

print(chan)
print(le)

W4A23
Nhập n
Cho S(k) = 1 + 2 + 3 + … + k
Tìm k sao cho S(k) lớn nhất nhưng nhỏ hơn n

In [ ]:
n = int(input())
s = 0
k = 0

while s + (k + 1) < n:
    k += 1
    s += k

print(k)

W4A24
Nhập vào A, tìm n nhỏ nhất sao cho
1 + 1/2 + 1/3 + 1/4 + ... + 1/n > A 


In [ ]:
A = float(input())
S = 0
n = 0

while S <= A:
    n += 1
    S += 1/n

print(n)

W4A25
Nhập vào một dãy số nguyên, ngưng nhập khi người dùng nhập -1.
Sau khi nhập xong, in số lớn nhất, số nhỏ nhất trong những số vừa nhập

In [11]:

numbers = []

print("Nhập vào các số nguyên (nhập -1 để dừng):")

while True:
    try:
        n = int(input())
        
        if n == -1:
            break
            
        numbers.append(n)
    except ValueError:
        print("Vui lòng chỉ nhập số nguyên!")

if len(numbers) > 0:
    max_val = max(numbers)
    min_val = min(numbers)
    
    print("--- Kết quả ---")
    print(f"Số lớn nhất: {max_val}")
    print(f"Số nhỏ nhất: {min_val}")
else:
    print("Bạn chưa nhập số nào hợp lệ trước khi nhấn -1.")

Nhập vào các số nguyên (nhập -1 để dừng):
Vui lòng chỉ nhập số nguyên!
Vui lòng chỉ nhập số nguyên!
--- Kết quả ---
Số lớn nhất: 48
Số nhỏ nhất: 2


W4A25
Nhập vào một dãy số nguyên, ngưng nhập khi người dùng nhập -1.
Sau khi nhập xong, in số lớn nhất, số nhỏ nhất trong những số vừa nhập

In [ ]:
A = int(input())
a, b = 1, 1
result = 1
while b <= A:
    result = b
    a, b = b, a + b
print(result)

w4a27
Nhập vào một chuỗi, hãy đếm xem trong chuỗi có bao nhiêu từ (quy định là chuỗi không có
ký tự đặc biệt, không số, không có dấu câu, chỉ có ký tự chữ và khoảng trắng)
Dùng cách: for trên index; KHÔNG dùng hàm split(), len()


In [ ]:
s = input()

count = 0
in_word = False

i = 0
while True:
    try:
        char = s[i]
        if char != ' ':
            if not in_word:
                count += 1
                in_word = True
        else:
            in_word = False
        i += 1
    except IndexError:
        break

print(count)

w4a28 Nhập vào một chuỗi, hãy in từ đầu tiên trong chuỗi
Dùng cách: for trên index; KHÔNG dùng hàm split() 


In [ ]:
chuoi = input("Nhập vào một chuỗi: ")

start_index = 0
for i in range(len(chuoi)):
    if chuoi[i] != ' ':
        start_index = i
        break

tu_dau_tien = ""
for i in range(start_index, len(chuoi)):
    if chuoi[i] == ' ':
        break
    tu_dau_tien += chuoi[i]

print("Từ đầu tiên là:", tu_dau_tien)

w4a29 Cho một chuỗi chỉ gồm các ký tự chữ cái thường và chữ cái hoa, hãy đổi ký tự hoa thành
thường và thường thành hoa. 

In [ ]:
s = input()
res = ""
for c in s:
    if c.islower():
        res += c.upper()
    else:
        res += c.lower()
print(res)

w4a30 Nhập vào một chuỗi có dạng 3 số nguyên, mỗi số nguyên cách nhau một dấu phẩy, hãy tính
tổng 3 số nguyên đó
VD:
Nhập: 3, 12, 15
Tổng: 30 

In [ ]:
s = input()
a, b, c = s.split(',')
print(int(a) + int(b) + int(c))

w4a31 Nhập vào một chuỗi, hãy đếm xem trong chuỗi có bao nhiêu ký tự in hoa, bao nhiêu ký tự in
thường, bao nhiêu ký tự số

In [ ]:
s = input()

chu_hoa = 0
chu_thuong = 0
chu_so = 0

for char in s:
    if char.isupper():
        chu_hoa += 1
    elif char.islower():
        chu_thuong += 1
    elif char.isdigit():
        chu_so += 1

print("-" * 30)
print(f"Số ký tự in hoa: {chu_hoa}")
print(f"Số ký tự in thường: {chu_thuong}")
print(f"Số ký tự số: {chu_so}")

### Homework 3

Bài 1: In các số dương lẻ bé hơn 100

In [ ]:
for i in range(1, 100):
    if i % 2 != 0:
        print(i, end=" ")

Bài 2: In các số dương chẵn chia hết cho 3 bé hơn 100

In [ ]:
for i in range(1, 100):
    if i % 2 == 0 and i % 3 == 0:
        print(i, end=" ")

Bài 3: Nhập vào số nguyên dương a, in ra bảng cửu chương của a

In [ ]:
a = int(input())
for i in range(1, 10):
    print(f"{a}*{i} = {a*i}")

Bài 4: Viết chương trình in ra hình tam giác cân có độ cao h được nhập từ bàn phím

In [ ]:
h = int(input())
for i in range(1, h+1):
    print(" " * (h-i), end="")
    print("*" * (2*i-1))


Bài 5: Nhập vào số nguyên dương n. Tính S = 1 + 2 + 3 + 4 + … + n

In [ ]:
n = int(input())
S = 0
for i in range(1, n+1):
    S += i
print(S)

Bài 6: Nhập vào số nguyên dương a, in ra tất cả các ước dương của a

In [ ]:
a = int(input())
for i in range(1, a+1):
    if a % i == 0:
        print(i, end=" ")

Bài 7: Nhập vào số nguyên dương a, đếm số ước dương của a

In [ ]:
a = int(input())
count = 0
for i in range(1, a+1):
    if a % i == 0:
        count += 1
print(count)

Bài 8: Nhập vào số nguyên dương a và b, in toàn bộ ước dương chung của a và b

In [ ]:
a, b = map(int, input().split())

for i in range(1, min(a, b) + 1):
    if a % i == 0 and b % i == 0:
        print(i, end=" ")

Bài 9:
Nhập vào số nguyên dương a, kiểm tra xem a có phải là số chính phương hay không? Số chính phương là số có căn bậc hai là một số nguyên

In [ ]:
a = int(input())
if int(a**0.5)**2 == a:
    print("True")
else:
    print("False") 

Bài 10:
Nhập vào số nguyên dương a, kiểm tra xem a có phải là số hoàn hảo hay không. Số hoàn hảo là số có tổng các ước (ước nhỏ hơn nó) bằng chính n

In [ ]:
a = int(input())
tong = 0
for i in range(1, a):
    if a % i == 0:
        tong += i
if tong == a:
    print("True")
else:
    print("False")

Bài 11:
Nhập vào số nguyên a, nếu nhập số âm thì yêu cầu nhập lại cho đến khi người dùng nhập đúng số dương. Nếu người dùng nhập đúng số dương thì in ra “Bạn nhập đúng quy tắc” và dừng chương trình.

In [ ]:
while True:
    a = int(input())
    if a<0:
        continue
    break
print("Ban nhap dung quy tac")

Bài 12:
Nhập số nguyên dương n, Cho S(k) = 1 + 2 + 3 + … + k. Tìm k sao cho S(k) lớn nhất nhưng nhỏ hơn n

In [ ]:
n = int(input())
S = 0
k = 0
while S<n:
    k += 1
    S += k
print(k-1)

Bài 13:
Nhập vào số nguyên dương A, tìm n nhỏ nhất sao cho
1 + 1/2 + 1/3 + 1/4 + ... + 1/n > A

In [ ]:
A = int(input())
n = 0
S = 0
while S<=A:
    n += 1
    S += 1/n
print(n)

Bài 14:
Nhập vào một dãy số nguyên, dừng nhập khi người dùng nhập -1. Sau khi nhập xong, in số lớn nhất, số nhỏ nhất trong những số vừa nhập. KHÔNG ĐƯỢC dùng mảng

In [ ]:
min = 0
max = 0
while True:
    n = int(input())
    if n != -1:
        if n>max:
            max = n
        if n<min:
            min = n
        continue
    break
print(max)
print(min)

Bài 15:
Nhập vào số nguyên dương n, đếm xem n có bao nhiêu chữ số. Làm bằng nhiều cách nhất có thể, ko dùng mảng.

In [ ]:
n = int(input())
count = 0
while n != 0:
    n //= 10
    count += 1
print(count)

Bài 16:
Nhập vào số nguyên dương n, đếm xem n có bao nhiêu chữ số chẵn, bao nhiêu chữ số lẻ

In [ ]:
n = int(input())
chan = 0
le = 0
while n != 0:
    chu_so = n % 10
    n //= 10
    if chu_so % 2 == 0:
        chan += 1
    else:
        le += 1
print(chan, "chan", le, "le")


Bài 17: Nhập vào số nguyên dương n, tính tổng các chữ số của n

In [ ]:
n = int(input())
tong = 0
while n != 0:
    tong += n%10
    n //= 10
print(tong)


Bài 18:
Nhập vào một số nguyên dương n, kiểm tra xem n có phải là số dạng 3^k hay không

In [ ]:
n = int(input())
while n % 3 == 0:
    n //= 3
if n == 1:
    print("True")
else:
    print("False")

Bài 19:
Dãy số fibonacci là dãy số được định nghĩa như sau: 1, 1, 2, 3, 5, 8, 13,... với số kế tiếp sẽ bằng tổng hai số trước đó.
Nhập vào số nguyên dương A, hãy tìm số trong dãy số fibonacci lớn nhất nhưng không vượt quá A

In [ ]:
A = int(input())
f1, f2 = 1, 1
while f1 + f2 <= A:
    f_next = f1 + f2
    f1 = f2
    f2 = f_next
print(f2)

Bài 20:
Nhập vào số nguyên dương n, đếm xem n có bao nhiêu chữ số là số nguyên tố

In [ ]:
import math
n = int(input())
count = 0
while n != 0:
    chu_so = n % 10
    n //= 10
    if chu_so < 2:
        count += 0
    else:
        for i in range(2, math.isqrt(chu_so) + 1):
            if chu_so % i == 0:
                count += 0
        count += 1
print(count)